### PR TITLE
Remove CONMAN_ADMIN_ROUTES

### PR DESCRIPTION
**Pull request by meshy**
Continuing from [this discussion](https://github.com/meshy/django-conman/pull/115#discussion_r141756485), we can avoid adding another setting if we "default to including all, but allow setting a whitelist or blacklist explicitly" to replace `CONMAN_ADMIN_ROUTES`.

The way I see it, there are two "bits" we need to get right here:

1. `RouteParentAdmin.get_child_models`. @Ian-Foote's [suggestion for getting all subclasses `__subclass__`](https://stackoverflow.com/questions/3862310/how-can-i-find-all-subclasses-of-a-class-given-its-name/33607093#33607093), seems like the way forward there.

1. The ability to override this. There are a number of options here:

    1. Have whitelist and blacklist settings to filter the result of the `__subclasses__` call above.

    1. Have a setting that overrides the call above with functionality similar to the existing `CONMAN_ADMIN_ROUTES`.

    1. Make `get_child_models` defer to a function, and allow that to be overridden with a setting / app config. This might fit nicely with the linked code because of the `yield` statement therein. We can just wrap it with a `list()`, and make sure we're always returning the same type of data regardless of whether we're overridden. This is the direction I'm currently thinking might be best.

    1. Other? I wouldn't be surprised if there's a better way...

@Ian-Foote, do you have any thoughts/ideas?

### CODE REVIEW

**Comment by meshy** [comment]
Tests failing due to an order of execution issue. I'll get back to this.

File: tests/routes/test_admin.py [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 
-        route_paths = ['redirects.RouteRedirect', 'redirects.URLRedirect']
-        with self.settings(CONMAN_ADMIN_ROUTES=route_paths):
-            models = admin.get_child_models()
+        with mock.patch.object(Route, 'get_subclasses') as get_subclasses:
+            get_subclasses.return_value = (c for c in [Route])
+            subclasses = admin.get_child_models()
 
-        self.assertEqual(models, [RouteRedirect, URLRedirect])
+        self.assertIsInstance(subclasses, list)
```
Comment (Ian-Foote) [inline tests/routes/test_admin.py]: Do we need to assert the mock is called too?

File: tests/routes/test_checks.py [hunk 2]
```diff
@@ -59,3 +61,49 @@ def test_unavailable(self):
...
+        """When a Route subclass is in the admin, show no error."""
+        errors = checks.subclasses_in_admin(app_configs=None)
+        self.assertEqual(errors, [])
+
+    def test_not_in_admin(self):
+        """When a Route subclass isn't in the admin, show an error."""
+        # Store the registered admin class for teardown.
+        admin_class = admin.site._registry[NestedRouteSubclass].__class__
+        try:
+            admin.site.unregister(NestedRouteSubclass)
```
Comment (Ian-Foote) [inline tests/routes/test_checks.py]: It probably doesn't matter in practice, but I think this should be outside the `try`.

File: tests/routes/test_checks.py [hunk 3]
```diff
@@ -59,3 +61,49 @@ def test_unavailable(self):
...
+            hint="Missing: {<class 'tests.models.NestedRouteSubclass'>}.",
+            id='conman.routes.E003',
+        )
+        self.assertEqual(result, [expected])
+
+    def test_admin_disabled(self):
+        """When the admin isn't active, don't force use of it."""
+        admin_class = admin.site._registry[NestedRouteSubclass].__class__
+
+        path = 'conman.routes.checks.apps.is_installed'
```
Comment (Ian-Foote) [inline tests/routes/test_checks.py]: Perhaps a comment to explain that `apps` is imported in `checks`, so we're actually mocking part of Django, not conman.

File: tests/routes/test_handlers.py [hunk 4]
```diff
@@ -148,30 +136,23 @@ class RouteViewHandlerCheckTest(TestCase):
...
         expected = Error(
-            'RouteWithNoView must have a `view` attribute.',
+            'RouteSubclass must have a `view` attribute.',
             hint='This is a requirement of RouteViewHandler.',
-            obj=RouteWithNoView,
+            obj=RouteSubclass,
         )
         self.assertEqual(errors, [expected])
 
     def test_function(self):
```
Comment (Ian-Foote) [inline tests/routes/test_handlers.py]: Perhaps `test_has_view` is a better name?

**Comment by meshy** [comment]
Thanks! `:)`

**Pull request merged**